### DISCUSSION TITLE
Renovate can not update Comment on PR

### DISCUSSION THREAD

**Discussion by Marvin0098** [renovatebot/renovate#23769] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, select which platform you are using.

AWS CodeCommit

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

When renovate adds a comment to a PR it cannot update the comment in the next run.

````
ERROR: Error ensuring PR (repository=repoName, baseBranch=develop, branch=renovate/develop-major-all-major)
93 | "err": {
94 | "name": "CommentNotCreatedByCallerException",
95 | "$fault": "client",
96 | "$metadata": {
97 | "httpStatusCode": 400,
98 | "requestId": "XXXXXX-XX-XX-XX-XXXXX",
99 | "attempts": 1,
100 | "totalRetryDelay": 0
101 | },
102 | "__type": "CommentNotCreatedByCallerException",
103 | "message": "You cannot update this comment.  The only person who can update a comment is the IAM user who created it.",
104 | "stack": "CommentNotCreatedByCallerException: You cannot update this comment.  The only person who can update a comment is the IAM user who created it.\n    at de_CommentNotCreatedByCallerExceptionRes (/usr/local/lib/node_modules/renovate/node_modules/@aws-sdk/client-codecommit/dist-cjs/protocols/Aws_json1_1.js:6285:23)\n    at de_UpdateCommentCommandError (/usr/local/lib/node_modules/renovate/node_modules/@aws-sdk/client-codecommit/dist-cjs/protocols/Aws_json1_1.js:5564:25)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at /usr/local/lib/node_modules/renovate/node_modules/@smithy/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24\n    at /usr/local/lib/node_modules/renovate/node_modules/@aws-sdk/middleware-signing/dist-cjs/awsAuthMiddleware.js:14:20\n    at /usr/local/lib/node_modules/renovate/node_modules/@smithy/middleware-retry/dist-cjs/retryMiddleware.js:27:46\n    at /usr/local/lib/node_modules/renovate/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26\n    at Object.updateComment (/usr/local/lib/node_modules/renovate/lib/modules/platform/codecommit/codecommit-client.ts:95:10)\n    at Proxy.ensureComment (/usr/local/lib/node_modules/renovate/lib/modules/platform/codecommit/index.ts:663:5)\n    at ensureComment (/usr/local/lib/node_modules/renovate/lib/modules/platform/comment.ts:16:17)\n    at processBranch (/usr/local/lib/node_modules/renovate/lib/workers/repository/update/branch/index.ts:841:13)\n    at writeUpdates (/usr/local/lib/node_modules/renovate/lib/workers/repository/process/write.ts:159:17)\n    at update (/usr/local/lib/node_modules/renovate/lib/workers/repository/process/extract-update.ts:214:11)\n    at Object.renovateRepository (/usr/local/lib/node_modules/renovate/lib/workers/repository/index.ts:75:19)\n    at attributes.repository (/usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/usr/local/lib/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /usr/local/lib/node_modules/renovate/lib/renovate.ts:18:22"
105 | }
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>


### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
`You cannot update this comment.  The only person who can update a comment is the IAM user who created it.`

**Comment by Marvin0098** [comment]
Yes, I have seen that the author is a different user. In my config there is this line:
`  "gitAuthor": "renovate@myCompany.com",`

But it looks like it is using a ID from CodeBuild, instead of that gitAuthor
![image](https://github.com/renovatebot/renovate/assets/86524330/7bd2d0c9-bb67-47fb-b7d6-c475b11a60f5)

**Reply by viceice** [reply]
codebuild probably sets git author and committer via environment variables, those override the renovate config. the same happens on most CI systems (Jenkins verified).

so simply unset those variables before running renovate.

**Reply by Marvin0098** [reply]
Thanks for the help.
Will try out the unset env variable tipp.

**Comment by Marvin0098** [comment]
I saw this issue: https://github.com/renovatebot/renovate/issues/19091

Is this the intended behaviour that renovate uses the CodeBuild ID, instead of `gitAuthor`? If so is this feature to update Comments broken?

**Reply by rarkins** [reply]
gitAuthor should be unrelated to comments, because it's only used for _commits_.

**Reply by Marvin0098** [reply]
Thanks for the quick answer, so I guess the changing Codebuild ID is the issue here? Dont know how to make that static.

**Reply by rarkins** [reply]
@PhilipAbed do you recall anything about this? I guess they need a static ID if they are to support editing comments? Either that or the ID might need elevated privileges to be able to edit anyone's comment

**Reply by PhilipAbed** [reply]
i remember the codebuild creates ARN depending on the build number, that was a problem when we wanted to filter by ARN.
see https://github.com/renovatebot/renovate/issues/19091#issuecomment-1333910537
i'm not sure how to solve that or how to make it static, the `authorArn` is automatically generated by amazon depending on where its coming from.
if u are in EC2 it will be an EC2 ARN, if its codebuild, it will be the codebuild number.

**Reply by PhilipAbed** [reply]
i tried :

```
phases:
  pre_build:
    commands:
      - git config user.name "xyz"
      - git config user.email "xyz@whitesourcesoftware.com"
      - git config --global credential.helper '!aws codecommit credential-helper $@'
      - git config --global credential.UseHttpPath true
  build:
    on-failure: CONTINUE
    commands:
      - npm install -g renovate
      - renovate
```

but still didnt change the author 
![image](https://github.com/renovatebot/renovate/assets/42116482/a5fd40cc-b9ae-4171-8d2f-c1b87474d20c)



![image](https://github.com/renovatebot/renovate/assets/42116482/c592475b-04c1-4bf5-945f-9ba1887bca82)
the committer is `Renovate Bot` but the PR author is always the ARN, unless you run from self hosted..
if you use any amazon resource, it will create its own ARN.

**Reply by Marvin0098** [reply]
Thanks for the testing, so it looks like when using CodeBuild there is no way to have an static ARN?

**Comment by PhilipAbed** [comment]
I tried changing the environment parameter: `CODEBUILD_BUILD_ARN`
but it doesnt allow me to.

I am not able to find environment variables for git author in AWS docs,
i tried setting 
`GIT_AUTHOR_NAME`
`GIT_AUTHOR_EMAIL`
`GIT_COMMITTER_NAME`
`GIT_COMMITTER_EMAIL`
but didn't work either

And we can't rely on the committers either according to [GetPullRequestResponse](https://docs.aws.amazon.com/codecommit/latest/APIReference/API_GetPullRequest.html)
they don't show the first/last commits not even with `GetBranchResponse`

And i'm out of ideas

The best thing i can do is post a question on AWS platform. [Question](https://repost.aws/questions/QUBZzjJdcSRIiPEBvjLAUAQw/changing-pr-author-from-codebuild).

**Comment by viceice** [comment]
@Met0z did you configure an explicit IAM user?

https://docs.renovatebot.com/modules/platform/codecommit/#iam-user

that's probably the only option to get a static arn?

**Reply by PhilipAbed** [reply]
that wouldn't work @viceice AWS he is working with `CodeBuild`(aws ci tool) and codebuild generates a new IAM for every build, so no matter what IAM Userhe logs in. the author will be the build number. need to learn how to change the git author in the ci tool.
the AWS documentation is vast and not clear, i wasn't able to find anything anywhere that could help.

according to AWS response https://repost.aws/questions/QUBZzjJdcSRIiPEBvjLAUAQw/changing-pr-author-from-codebuild
they said the "get commit" might help, but i didnt try it cuz i dont think that helps our case
